src/state/slices/tradeQuoteSlice/selectors.ts

```diff
@@ -565,7 +565,7 @@ export const selectTradeQuoteAffiliateFeeAfterDiscountUsd = createSelector(
     }),
   selectActiveQuoteAffiliateBps,
   (calculatedFees, affiliateBps) => {
-    if (!affiliateBps) return
+    if (!affiliateBps || affiliateBps === '0') return '0'
```

Comment:
Two remarks:

1. Can we return `bn(0)` instead of '0', as the return type is now `BigNumber | '0'` vs. `BigNumber | undefined` previously, feels weird to have two types 

<img width="582" alt="image" src="https://github.com/user-attachments/assets/6b496533-3a68-4161-96fb-bc6607ac7517" />

2. regardless of whether or not we return a '0' string literal or a consistent BN vs. BN | undefined, we would absolutely need to update things at types level and runtime across the board to reflect this if we go this route i.e:

https://github.com/shapeshift/web/blob/b84dfa49fc42a0489817017c0002829804b9c452/src/state/slices/tradeQuoteSlice/selectors.ts#L573-L580

-  `!tradeAffiliateFeeAfterDiscountUsd` check will always evaluate to false here, as `Boolean(0) === false` but `Boolean('0') == true` so we're changing the heuristics here
- With that bein said, there is no reason to have the early return altogether anymore if we go this route, since neither of `tradeAffiliateFeeAfterDiscountUsd` nor `sellUserCurrencyRate` can be falsy

https://github.com/shapeshift/web/blob/b84dfa49fc42a0489817017c0002829804b9c452/src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/FeeStep.tsx#L63
https://github.com/shapeshift/web/blob/b84dfa49fc42a0489817017c0002829804b9c452/src/components/MultiHopTrade/components/TradeConfirm/TradeConfirmFooterContent/TradeConfirmSummary.tsx#L215

 
- This doesn't need to be a `bnOrZero()` anymore as this will never be undefined after removing the early return in `selectTradeQuoteAffiliateFeeAfterDiscountUserCurrency`

https://github.com/shapeshift/web/blob/b84dfa49fc42a0489817017c0002829804b9c452/src/components/MultiHopTrade/components/SharedTradeInput/SharedTradeInputFooter/SharedTradeInputFooter.tsx#L16

- Can be `string` instead of `string | undefined`


@NeOMakinG what's the rationale here? The type change here feels like a one-off which should be handled across the board, but I fail to see the reason this was changed, since we don't handle said change